src/wfs/src/main/java/org/geoserver/wfs/FeatureSizeFeatureCollection.java

```diff
@@ -0,0 +1,158 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wfs;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.geotools.data.FeatureSource;
+import org.geotools.data.Query;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.feature.collection.DecoratingSimpleFeatureCollection;
+
+import org.opengis.feature.Feature;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.type.FeatureType;
+
+
+/**
+ * A feature collection which cache features when the feature source 
```

Comment:
"which caches a configurable (and small) amount of features"

Reply:
Thanks @aaime, I changed it